client/finality-grandpa/src/communication/gossip.rs

```diff
@@ -539,21 +542,26 @@ impl<N: Ord> Peers<N> {
 	}
 
 	fn authorities(&self) -> usize {
-		self.inner.iter().filter(|(_, info)| info.roles.is_authority()).count()
+		self.inner.iter().filter(|(_, info)| matches!(info.roles, ObservedRole::Authority)).count()
 	}
 
 	fn non_authorities(&self) -> usize {
-		self.inner.iter().filter(|(_, info)| !info.roles.is_authority()).count()
+		self.inner
+			.iter()
+			.filter(|(_, info)| matches!(info.roles, ObservedRole::Full | ObservedRole::Light))
+			.count()
 	}
 
 	fn reshuffle(&mut self) {
 		let mut lucky_peers: Vec<_> = self.inner
 			.iter()
-			.filter_map(|(id, info)| if !info.roles.is_authority() { Some(id.clone()) } else { None })
+			.filter_map(|(id, info)|
+				if matches!(info.roles, ObservedRole::Full | ObservedRole::Light) { Some(id.clone()) } else { None })
 			.collect();
 		let mut lucky_authorities: Vec<_> = self.inner
 			.iter()
-			.filter_map(|(id, info)| if info.roles.is_authority() { Some(id.clone()) } else { None })
+			.filter_map(|(id, info)|
+				if matches!(info.roles, ObservedRole::Authority) { Some(id.clone()) } else { None })
```

Comment:
```suggestion
				if matches!(info.roles, ObservedRole::Authority | ObservedRole::OurGuardedAuthority) { Some(id.clone()) } else { None })
```

or the introduce the `role.is_authority` function.